### PR TITLE
use schema lims

### PR DESCRIPTION
**Pull request by moonso**
## Description
This PR adds a pydantic model for lims sample and give an example on how to utilize that in meta/orders/lims

### How to test
- [ ] Tests are automated

### Expected test outcome
- [ ] check that automated tests pass
- [ ] Take a screenshot and attach or copy/paste the output.

## Review
- [ ] code approved by
- [ ] tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: cg/meta/orders/lims.py [hunk 1]
```diff
@@ -1,65 +1,32 @@
...
-                        "tumour": sample.get("tumour") or False,
-                        "tumour_purity": sample.get("tumour_purity"),
-                        "volume": sample.get("volume"),
-                        "well_position_rml": sample.get("well_position_rml"),
-                        "verified_organism": sample.get("verified_organism"),
-                    },
-                }
-            )
-        return samples_lims
+def to_lims(customer: str, samples: List[dict]) -> List[LimsSample]:
```
Comment (Mropat) [inline cg/meta/orders/lims.py:10]: This function can be renamed! build_lims_sample could could be a suggestion

File: cg/meta/orders/lims.py [hunk 2]
```diff
@@ -1,65 +1,32 @@
...
-        """Process samples to add them to LIMS."""
-        samples_lims = self.to_lims(data["customer"], samples)
-        project_name = data["ticket"] or data["name"]
-        project_data = self.lims.submit_project(project_name, samples_lims)
-        lims_map = self.lims.get_samples(projectlimsid=project_data["id"], map_ids=True)
-        return project_data, lims_map
+def process_lims(lims_api: LimsAPI, data: dict, samples: List[dict]):
+    """Process samples to add them to LIMS."""
+    samples_lims: List[LimsSample] = to_lims(data["customer"], samples)
+    project_name = data["ticket"] or data["name"]
```
Comment (Mropat) [inline cg/meta/orders/lims.py:26]:
```suggestion
    project_name = data.get("ticket") or data["name"]
```

File: cg/meta/orders/lims.py [hunk 3]
```diff
@@ -1,65 +1,32 @@
...
+    # Create new lims project
+    project_data = lims_api.submit_project(
+        project_name, [lims_sample.dict() for lims_sample in samples_lims]
```
Comment (Mropat) [inline cg/meta/orders/lims.py:29]: would be cool with kwargs for added readability

**Review by Mropat — APPROVED** [review]

File: cg/meta/orders/lims.py (hunk 1 above)
Comment (moonso) [reply cg/meta/orders/lims.py:10]: changed

File: cg/meta/orders/lims.py (hunk 2 above)
Comment (moonso) [reply cg/meta/orders/lims.py:26]: Fixed

File: cg/meta/orders/api.py [hunk 4]
```diff
@@ -69,7 +69,9 @@ def _submit_rml(self, order: dict) -> dict:
 
     def _submit_pools(self, order):
         status_data = self.pools_to_status(order)
-        project_data, lims_map = self.process_lims(order, order["samples"])
+        project_data, lims_map = process_lims(
+            lims_api=self.lims, data=order, samples=order["samples"]
```
Comment (patrikgrenfeldt) [inline cg/meta/orders/api.py:73]: Why send both the full order and part of the order (order["samples"])?
I suggest we find ourselves a more descriptive argument name than "data"

**Review by patrikgrenfeldt — APPROVED** [review]
Great, had a couple of suggestions 👍

File: cg/meta/orders/api.py (hunk 4 above)
Comment (moonso) [reply cg/meta/orders/api.py]: Good suggestions, changed accordingly

**Comment by moonso** [comment]
Screen Shot 2021-03-22 at 11 14 29

Deployed on hasta

**Pull request merged**